examples/mcmc.jl

```diff
@@ -90,17 +138,28 @@ function data_generating_procedure(t::Int64)
         Gen.set_submap!(obs, addr, _choices)
     end
 
-    return (gargs, obs)
+    #=  plt = lineplot(1:t, get_zs(trace),
+                   title="Height of ball", name="observation with res $(trace[:prior => 1 => :restitution])",
+                   xlabel="t", ylabel="z", canvas=DotCanvas,
+                   border=:ascii) =#
+    plt = plot(1:t, get_zs(trace),title="Height of ball", xlabel="t", ylabel="z",label="Observation")
+    
+    return (gargs, obs, plt)
 
 end
 
 function main()
 
     t = 60 # 1 second of observations
-    (gargs, obs) = data_generating_procedure(t)
-
-    (tr, aratio) = inference_procedure(gargs, obs)
-
+    (gargs, obs, plt) = data_generating_procedure(t)
+    display(plt)
+    
+    update_vis(trace) = display(plot!(plt, 1:t, get_zs(trace), label="res $(trace[:prior => 1 => :restitution])"))
+    (tr, aratio) = inference_procedure(gargs, obs, update_vis)
+    print(tr[:prior => 1])
+
+    println("press enter to exit the program")
+    readline()
```

Comment:
why is this here?

Reply:
i added it because on my machine the plots close when the application finishes

Reply:
ah, ok. i'm not the biggest fan and would prefer that you just write to disk but that's more of a style thing